### PR TITLE
Initial layout for documentation

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
on the internals of the VB.net language, Compiler and its related Tools.

### CODE REVIEW

**Comment by gafter** [comment]
Compiler documentation does not belong in the language repo.

**Comment by AdamSpeight2008** [comment]
@gafter It will be more of guidance and advice, the official documentation will still on the Roslyn repo.

**Comment by AdamSpeight2008** [comment]
@gafter 
We could pull some to documents out of `Roslyn` into there respective lang repo.
The language specification is already in both of them.

**Comment by KathleenDollard** [comment]
@AdamSpeight2008 Can you please open an issue where we can discuss the layout of this and how it is distinct from the Roslyn docs and the general Visual Basic documentation. You can reference this PR. 

Roslyn is Open Source so everyone can fork and play with the repos. But I don't people to misunderstand. Coding a solution for their favorite feature does not make it more likely that it will be accepted into the language.

File: repo-tags.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+| `[LDM:Declined]`        | Red    | Proposal was Declinec by the LDM with rationale. |
+| `[Withdrawn]`           | Red    | Contributor has withdrawn the proposal |
+| `[Up for Grabs ]`       |        | An accepted proposal pending a developer to implement feature. Used in association with tag `[Up for Grabs]` so an interested party could develop. |
+| `[Feature]`             |        | A proposal that has been accepted be the LDM |
+| `[Feature:Development]` |        | Feature is in development into the official language. |
+| `[Feature:Completed]`   |        | Feature has been developed. |
+| `[Feature:Pending]`     |        | Feature is pending review / sign off  |
+| `[Feature:Preview]`     | Silver | Feature is now in an official preview version (eg CTP) of the language |
+| `[Feature:Released]`    | Gold   |  Feature is now in an official release version of the language. If an external contributor does manage to achieve this, *(I think)* we as a community should acknowledge them with something special. |
+
```
Comment (KathleenDollard) [inline repo-tags.md]: @sharwell Are these tags similar to ones in other repos?

**Comment by KathleenDollard** [comment]
@AdamSpeight2008 I think this single PR is going to make it nearly impossible to approve. Can you please split this in to separate PRs. You can combine the internals, but I think the others need a separate PR to facilitate discussion and resolution. 

Why aren't the compiler switches elsewhere we can reference them? Having anything documented here that isn't the point of truth seems like a bad idea. 

What is the point of the internals section? If it's an introduction to Roslyn, I think a community website on this might be a better location (it can be updated without Microsoft approval, among other things). 

The tags will need specific discussion, and that doc clearly belongs in this repo. Did you find something to start from? It looks fairly complete.

**Comment by AdamSpeight2008** [comment]
@KathleenDollard 
> Did you find something to start from? It looks fairly complete. 

The content was partially reverse engineered (as source documentation is lacking) 
>What is the point of the internals section? If it's an introduction to Roslyn, I think a community website on this might be a better location (it can be updated without Microsoft approval, among other things).

Not just on the usage of Roslyn but the internal code used with Roslyn source.

I also think we should extract the documentation and examples out of the roslyn repo into its own repo, then people can contribute to doc with pulling the code.

**Comment by gafter** [comment]
This proposed documentation appears to be about Roslyn, not the VB language specification. As such it is out of place in this repository.

**Pull request closed without merging**